src/library/clementine/clementineplaylistmodel.cpp

```diff
@@ -0,0 +1,299 @@
+                if (!query.exec()) {
+                    LOG_FAILED_QUERY(query);
+                }
+                // qDebug() << "-----" << entry.pTrack->title << query.executedQuery();
+            }
+
+            endInsertRows();
+        }
+    }
+
+    QStringList tableColumns;
+    tableColumns << CLM_VIEW_ORDER // 0
+                 << CLM_PREVIEW;
+
+    QStringList trackSourceColumns;
+    trackSourceColumns << CLM_VIEW_ORDER // 0
+                       << CLM_ARTIST
+                       << CLM_TITLE
+                       << CLM_DURATION
+                       << CLM_URI
+                       << CLM_ALBUM
+                       << CLM_ALBUM_ARTIST
+                       << CLM_YEAR
+                       << CLM_RATING
+                       << CLM_GENRE
+                       << CLM_GROUPING
+                       << CLM_TRACKNUMBER
+                       //<< CLM_DATEADDED
+                       << CLM_BPM
+                       << CLM_BITRATE
+                       << CLM_COMMENT
+                       << CLM_PLAYCOUNT
+                       << CLM_COMPOSER;
+
+    QSharedPointer<BaseTrackCache> trackSource(
+            new BaseTrackCache(m_pTrackCollectionManager->internalCollection(),
+                    Clementine_TABLE,
+                    CLM_VIEW_ORDER,
+                    trackSourceColumns,
+                    false));
+
+    setTable(Clementine_TABLE, CLM_VIEW_ORDER, tableColumns, trackSource);
+    setSearch("");
+    setDefaultSort(fieldIndex(PLAYLISTTRACKSTABLE_POSITION), Qt::AscendingOrder);
+    setSort(defaultSortColumn(), defaultSortOrder());
+}
+
+TrackModel::Capabilities ClementinePlaylistModel::getCapabilities() const {
+    return Capability::AddToTrackSet |
+            Capability::AddToAutoDJ |
+            Capability::LoadToDeck |
+            Capability::LoadToSampler;
+}
+
+Qt::ItemFlags ClementinePlaylistModel::flags(const QModelIndex& index) const {
+    return readWriteFlags(index);
+}
+
+TrackId ClementinePlaylistModel::doGetTrackId(const TrackPointer& pTrack) const {
+    if (pTrack) {
+        for (int row = 0; row < rowCount(); ++row) {
+            const QUrl rowUrl(getFieldString(index(row, 0), CLM_URI));
+            if (TrackFile::fromUrl(rowUrl) == pTrack->getFileInfo()) {
+                return TrackId(getFieldVariant(index(row, 0), CLM_VIEW_ORDER));
+            }
+        }
+    }
+    return TrackId();
+}
+
+QVariant ClementinePlaylistModel::getFieldVariant(const QModelIndex& index,
+        const QString& fieldName) const {
+    return index.sibling(index.row(), fieldIndex(fieldName)).data();
+}
+
+QString ClementinePlaylistModel::getFieldString(const QModelIndex& index,
+        const QString& fieldName) const {
+    return getFieldVariant(index, fieldName).toString();
+}
+
+TrackPointer ClementinePlaylistModel::getTrack(const QModelIndex& index) const {
+    QString location = getTrackLocation(index);
+
+    if (location.isEmpty()) {
+        // Track is lost
```

Comment:
Here a debug log in the log file would actually be helpful.

Reply:
added this in https://github.com/mixxxdj/mixxx/pull/3173/commits/f2a59de119ecefa97cf8d9ae8821a79d37650a05